src/MergeContext.js

```diff
@@ -695,6 +702,7 @@ class PullRequest {
         let labels = await GH.getLabels(this._prNumber());
         assert(!this._labels);
         this._labels = new Labels(labels, this._prNumber());
+        this._stagingFailed = this._labels.has(Config.failedStagingChecksLabel());
```

Comment:
Why do we need to set this._stagingFailed based on the label at all? Why not always compute its value from scratch, based on staged commit history/state? AFAICT, this complication is rooted in a case where there is no staged commit anymore, but our _recalculateStagingFailed() handles that exact situation already! Perhaps this is just an optimization that is meant to reduce the number of _recalculateStagingFailed() calls?

Reply:
After some consideration, I decided to not compute _stagingFailed from the label as you suggested. Done at cde4f45.